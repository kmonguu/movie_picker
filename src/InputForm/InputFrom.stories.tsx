import { ComponentMeta, ComponentStory } from '@storybook/react'
import React from 'react'
import InputForm from '.'

const Template: ComponentStory<typeof InputForm> = () => {
  return <InputForm />
}

export const Default = Template.bind({})

export default {
  title: 'Component/InputForm',
  component: InputForm,
} as ComponentMeta<typeof InputForm>
