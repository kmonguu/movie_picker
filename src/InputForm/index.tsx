/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react'

interface InputFormProps {
  handleRandomNumber: () => void
  value: any
  onChange: any
}

function InputForm({ handleRandomNumber, value, onChange }: InputFormProps) {
  return (
    <div css={wrapperCss}>
      <input placeholder="영화제목을 맞춰보세요!" value={value} onChange={onChange} />
      <button onClick={handleRandomNumber}>저장</button>
    </div>
  )
}

const wrapperCss = css`
  display: flex;
  justify-content: center;
  align-items: center;

  > input {
    margin: 40px 27px 40px 0;
    padding: 5px;
  }

  > button {
    width: 50px;
    height: 30px;

    border: none;
    border-radius: 5px;
    background: #5c829c;
    color: #fff;

    :hover {
      background: #53748a;
      cursor: pointer;
    }
  }
`

export default InputForm
