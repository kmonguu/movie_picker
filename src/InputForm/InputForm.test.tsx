import React from 'react'
import { render } from '@testing-library/react'
import InputForm from '.'

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const profile = render(<InputForm username="kimNaYeon" age="23" />)
    expect(profile.container).toMatchSnapshot()
  })
  it('shows the props correctly', () => {
    const utils = render(<InputForm username="kimNaYeon" age="23" />)
    utils.getByText('kimNaYeon')
    utils.getByText('23')
  })
})
