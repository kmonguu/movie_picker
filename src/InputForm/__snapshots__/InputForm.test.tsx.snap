// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <span>
      kimNaYeon
    </span>
    Â 
    <span>
      23
    </span>
  </div>
</div>
`;
